### PR TITLE
[stable/cockroachdb]: update version to 20.1.1

### PR DESCRIPTION
**Pull request by dt**
Signed-off-by: David Taylor <tinystatemachine@gmail.com>
#### Is this a new chart

No

#### What this PR does / why we need it:
Bump version of cockroachdb to v20.1.1

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [ ] ~Variables are documented in the README.md~
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by DuskEagle** [comment]
Looks like an instance of https://github.com/helm/helm/issues/3480. Going to retry the test.

/retest

**Comment by DuskEagle** [comment]
This is the same issue we ran into on #21904. When running the chart tests outside of prow.k8s.io, the tests pass fine. Even within prow, the actual CockroachDB cluster is able to launch successfully. 

However, it seems Helm has a problem contacting Tiller early on to get the deployment information, and that is causing the test to report as failed. See https://kubernetes.slack.com/archives/CHK6WB5R8/p1587498416035000.

I'm kind of stumped on this, as I can't get a repro locally. My next step will be to check that I am running the same Helm version locally as on the server, to see if that could be the issue.

**Comment by DuskEagle** [comment]
Tested using Helm v2.9.1, which appears to be the version of Helm used by prow.k8s.io for these tests. Still can't reproduce this failure locally.

File: stable/cockroachdb/Chart.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
 name: cockroachdb
 home: https://www.cockroachlabs.com
-version: 3.0.7
-appVersion: 19.2.5
+version: 3.0.8
```
Comment (cpanato) [inline stable/cockroachdb/Chart.yaml]:
```suggestion
version: 4.0.0
```
it is a major bump in the app version, need reflect in the chart version

also if users upgrade from 19.xxx to 20.xx there are any breaking changes?

Comment (dt) [reply stable/cockroachdb/Chart.yaml]: There are some backwards-incompatible changes to some of the SQL functions documented in the release notes here: https://www.cockroachlabs.com/docs/releases/v20.1.0.html#backward-incompatible-changes -- is there somewhere I should be linking that?

Comment (cpanato) [reply stable/cockroachdb/Chart.yaml]: Please link in the Readme and make a section to describe how to upgrade existing deployments to this new release

**Comment by ericharmeling** [comment]
@dt 
We can probably close this in favor of https://github.com/helm/charts/pull/22713.

**Comment by dt** [comment]
Closing in favor of #22536 which updates documentation as well.

**Pull request closed without merging**